### PR TITLE
Migration of Epoch into separate repository

### PR DESCRIPTION
**Pull request by a-sr**
This PR factors out all Eclipse (Epoch) specific projects and restructures the remaining source files such that they can be used as a sub module. The new repository is here: https://github.com/lf-lang/epoch
Implements parts of #1276.

Involved tasks (LF repo):
- [x] Move contents of `org.lflang.diagram` and  `org.lflang.ide` to `org.lflang`
- [x] Remove all Epoch related projects
- [x] Move `EclipseErrorRepoter` to Epoch repo
- [x] Adjust mwe build configuration
- [x] Document mwe build configuration implications (@a-sr)
- [x] Adjust Gradle build
- [x] Adjust GH build actions
- [x] Fix Gradle tests

Involved tasks (Epoch repo):
- [x] Include lingua-franca repository as submodule
- [x] Adjust Maven build
- [x] Adjust mwe build configuration
- [x] Setup GH build actions
- [x] Adjust Oomph setup
- [x] (not possible) Reduce redundancy in mwe configurations
- [x] Fix eclipse test (template check)
- [x] Implement Epoch in-tool compilation test
- [ ] Set repo to public 

Remaining tasks:
- https://github.com/lf-lang/vscode-lingua-franca/issues/100
- https://github.com/lf-lang/lingua-franca/issues/1613
- https://github.com/lf-lang/lingua-franca/issues/1614
- https://github.com/lf-lang/epoch/issues/3
- https://github.com/lf-lang/lf-lang.github.io/issues/128

### CODE REVIEW

**Comment by a-sr** [comment]
@cmnrd, @lhstrh I'd say the bulk of the work for the restructuring is done now. The Gradle build in LF works and likewise does the maven build in the new Epoch repo. Everything is as before, just in separate repos.
I still have a few open ToDos (see PR description) and I would appreciate if you could help me with these.
My current focus is to wrap things up in the new Epoch repo. Hence, it would be great if one of you could take a look at the failing Gradle tests. After I moved the `diagrams` and `ide` plugins into `org.lflang`, suddenly the majority of JUnit test run by Gradle reported a `SecurityException`. I did not yet further investigated the stacktrace.

Btw, the `build` job in GH only fails for this PR because is uses the configuration from the master branch. Since I only removed the maven build commands, the GH configuration should be fine.

**Comment by lhstrh** [comment]
This looks like fantastic progress, @a-sr! Thank you so much for doing this. It looks soooo much cleaner. I can pitch in with the remaining work. Perhaps @petervdonovan could with the adjustments that will be necessary in the VS Code repo.

File: org.lflang.tests/build.gradle [hunk 1]
```diff
@@ -1,3 +1,10 @@
+repositories {
+    mavenCentral()
+    // TODO Remove this in-official maven repository as soon as Klighd is released to maven central in the future.
```
Comment (lhstrh) [inline org.lflang.tests/build.gradle:3]:
```suggestion
    // TODO Replace this unofficial maven repository as soon as Klighd is released to maven central in the future.
```

File: org.lflang/build.gradle [hunk 2]
```diff
@@ -1,8 +1,16 @@
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
+repositories {
+    mavenCentral()
+    // TODO Remove this in-official maven repository as soon as Klighd is released to maven central in the future.
```
Comment (lhstrh) [inline org.lflang/build.gradle:5]:
```suggestion
    // TODO Replace this unofficial maven repository as soon as Klighd is released to maven central in the future.
```

**Review by lhstrh — COMMENTED** [review]
This all looks great! Thanks again for doing this. Let's hold off the merge for now, but I'll make it a priority to get it merged very soon.

File: org.lflang.tests/build.gradle [hunk 3]
```diff
@@ -1,3 +1,10 @@
+repositories {
+    mavenCentral()
+    // TODO Replace this unofficial maven repository as soon as Klighd is released to maven central in the future.
```
Comment (lhstrh) [inline org.lflang.tests/build.gradle:3]: #1615

File: org.lflang/build.gradle [hunk 4]
```diff
@@ -1,8 +1,16 @@
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
+repositories {
+    mavenCentral()
+    // TODO Replace this unofficial maven repository as soon as Klighd is released to maven central in the future.
+    maven {
+        url "https://rtsys.informatik.uni-kiel.de/~kieler/files/repo/"
```
Comment (lhstrh) [inline org.lflang/build.gradle:7]: #1615

**Review by lhstrh — APPROVED** [review]

**Comment by petervdonovan** [comment]
This branch deletes the `generateLanguageDiagramServer` Gradle task. How am I supposed to build the language server now?

**Comment by lhstrh** [comment]
Is it https://github.com/lf-lang/vscode-lingua-franca/issues/100?

**Pull request merged**